app/controllers/api/vcs.php

```diff
@@ -593,32 +611,100 @@
         $files = \array_column($files, 'name');
         $languages = $github->listRepositoryLanguages($owner, $repositoryName);
 
-        $detectorFactory = new Detector($files, $languages);
+        $detector = new Packager($files);
+        $detector
+            ->addOption(new Yarn())
+            ->addOption(new PNPM())
+            ->addOption(new NPM());
+        $detectedPackager = $detector->detect();
 
-        $detectorFactory
-            ->addDetector(new JavaScript())
-            ->addDetector(new Bun())
-            ->addDetector(new PHP())
-            ->addDetector(new Python())
-            ->addDetector(new Dart())
-            ->addDetector(new Swift())
-            ->addDetector(new Ruby())
-            ->addDetector(new Java())
-            ->addDetector(new CPP())
-            ->addDetector(new Deno())
-            ->addDetector(new Dotnet());
+        $packagerName = $detectedPackager ? $detectedPackager->getName() : 'npm';
 
-        $runtime = $detectorFactory->detect();
+        $detection = [];
 
-        $runtimes = Config::getParam('runtimes');
-        $runtimeDetail = \array_reverse(\array_filter(\array_keys($runtimes), function ($key) use ($runtime, $runtimes) {
-            return $runtimes[$key]['key'] === $runtime;
-        }))[0] ?? '';
+        if ($type === 'framework') {
+            $detection = [
+                'framework' => '',
+                'installCommand' => '',
+                'buildCommand' => '',
+                'outputDirectory' => '',
+            ];
+
+            $frameworkDetector = new Framework($files, $packagerName);
+            $frameworkDetector
+                ->addOption(new Flutter())
+                ->addOption(new Nuxt())
+                ->addOption(new Astro())
+                ->addOption(new Remix())
+                ->addOption(new SvelteKit())
+                ->addOption(new NextJs());
+
+            $detectedFramework = $frameworkDetector->detect();
+
+            if ($detectedFramework) {
+                $framework = $detectedFramework->getName();
+                $detection['installCommand'] = $detectedFramework->getInstallCommand();
+                $detection['buildCommand'] = $detectedFramework->getBuildCommand();
+                $detection['outputDirectory'] = $detectedFramework->getOutputDirectory();
+            }
 
-        $detection = [];
-        $detection['runtime'] = $runtimeDetail;
+            if (!empty($framework)) {
+                $frameworks = Config::getParam('frameworks');
+                $frameworkDetail = \array_reverse(\array_filter(\array_keys($frameworks), function ($key) use ($framework, $frameworks) {
+                    return $frameworks[$key]['key'] === $framework;
+                }))[0] ?? '';
+                $detection['framework'] = $frameworkDetail;
```

Comment:
Lets also throw exception here, when we cant find framework

Reply:
You can probably trigger this on purpose by changing .env, and removing framework from list of allowed frameworks